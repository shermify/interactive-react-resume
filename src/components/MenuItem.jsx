import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-scroll';

const MenuItem = props => {
  const linkProps = {
    ...props,
    className: 'nav-link',
    duration: 500,
    smooth: true,
    role: 'menuitem',
    tabIndex: '0',
  };
  return (
    <li className="nav-item">
      <Link {...linkProps}>{props.children}</Link>
    </li>
  );
};

MenuItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node,
};

MenuItem.defaultProps = {
  children: '',
};

export default MenuItem;
