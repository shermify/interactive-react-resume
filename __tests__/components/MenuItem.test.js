import React from 'react';
import renderer from 'react-test-renderer';
import MenuItem from '../../src/components/MenuItem';

describe('<MenuItem />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MenuItem to="test">Test Link</MenuItem>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
