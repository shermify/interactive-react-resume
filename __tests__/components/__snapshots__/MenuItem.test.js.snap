// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> renders correctly 1`] = `
<li
  className="nav-item"
>
  <a
    className="nav-link"
    onClick={[Function]}
    role="menuitem"
    tabIndex="0"
  >
    Test Link
  </a>
</li>
`;
